src/vs/workbench/contrib/preferences/browser/preferencesWidgets.ts

```diff
@@ -314,6 +323,19 @@ export class SettingsTargetsWidget extends Widget {
 		}
 	}
 
+	updateLanguageFilterIndicators(filter: string | undefined) {
+		this.resetLabels();
+		if (filter) {
+			const languageToUse = this.languageService.getLanguageName(filter);
+			if (languageToUse) {
+				this.userLocalSettings.label += ` [${languageToUse}]`;
+				this.userRemoteSettings.label += ` [${languageToUse}]`;
```

Comment:
I guess an issue here is that we might end up with Remote [hostLabel] [languageToUse] instead of Remote [hostLabel, languageToUse].